projects/angular/src/forms/number-input/number-input.ts

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2016-2024 Broadcom. All Rights Reserved.
+ * The term "Broadcom" refers to Broadcom Inc. and/or its subsidiaries.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {
+  Directive,
+  ElementRef,
+  HostListener,
+  Injector,
+  Optional,
+  Renderer2,
+  Self,
+  ViewContainerRef,
+} from '@angular/core';
+import { NgControl } from '@angular/forms';
+
+import { FocusService } from '../common/providers/focus.service';
+import { WrappedFormControl } from '../common/wrapped-control';
+import { ClrNumberInputContainer } from './number-input-container';
+
+@Directive({
+  selector: 'input[type="number"][clrNumberInput]',
+  host: { '[class.clr-input]': 'true', '[class.clr-number-input]': 'true' },
+})
+export class ClrNumberInput extends WrappedFormControl<ClrNumberInputContainer> {
+  protected override index = 1;
+
+  constructor(
+    @Optional() private focusService: FocusService,
+    vcr: ViewContainerRef,
+    injector: Injector,
+    @Self()
+    @Optional()
+    private control: NgControl,
+    renderer: Renderer2,
+    el: ElementRef<HTMLInputElement>
```

Comment:
```suggestion
    protected override el: ElementRef<HTMLInputElement>
```

If you override the `el` property here, you won't need to assert the `nativeElement` type below.